### PR TITLE
Enable CA verification

### PR DESCRIPTION
**Pull request by wscalf**
This PR naively enables CA verification in the grpc client to SpiceDB.
According to the [documentation](https://pkg.go.dev/github.com/authzed/grpcutil#WithSystemCerts), grpcutil.WithSystemCerts() already handles finding the default root CA certs for the system, which theoretically should include the generated root CA cert for OpenShift, so it seems likely this just works, and I think it'd be good to try it.

If it _doesn't_ just work, we should probably backlog it, but if it does, hey, easy win.

### CODE REVIEW

File: infrastructure/repository/authzed/SpiceDbAccessRepository.go [hunk 1]
```diff
@@ -331,7 +331,7 @@ func (s *SpiceDbAccessRepository) NewConnection(spiceDbEndpoint string, token st
 	if !useTLS {
 		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	} else { // Use TLS - but for now SKIP CA verification
```
Comment (Rajagopalan-Ranganathan) [inline infrastructure/repository/authzed/SpiceDbAccessRepository.go:333]: Update the comment to relfect the chnges

Comment (wscalf) [reply infrastructure/repository/authzed/SpiceDbAccessRepository.go]: Ah, while trying to keep it as minimal as possible, I skipped right over that. I missed that the variable was still called skipCA too. Updated both.

**Comment by Rajagopalan-Ranganathan** [comment]
LGTM - When we merge this, we need to be around for downstream deployment. If there is a connection issue, the service is broken. Should be ready to revert. And debug later.

**Comment by Rajagopalan-Ranganathan** [comment]
> LGTM - When we merge this, we need to be around for downstream deployment. If there is a connection issue, the service is broken. Should be ready to revert. And debug later.

**Comment by Rajagopalan-Ranganathan** [comment]
@wscalf  - I am thinking out loud, should this is be made a flag as well? So that we don't need to deploy a new image when things go south downstream, rather just update the deployment template? So enable CA verification or not? Given it can take close to 30 mins for the full cycle, WDYT?

**Comment by wscalf** [comment]
If the goal is to minimize downtime if it doesn't work, I think we have multiple options for that already: we could use the imagestream to roll back immediately (by decrementing the generation field) and then do a new deploy when ready, and we could also use that oc import-image command to speed up the roll forward.

There may be something to be said for letting upstream users run unsecure TLS as a workaround, but I can't imagine anyone would want to do that- they'd be doing it because there were issues with their PKI. And, now that you have me thinking about it, I wonder if there's anything we could do to help troubleshoot that, like maybe logging the root CA cert presented by the server and any available info about what system certs there are and where they were loaded from, stuff that may help a user troubleshoot their PKI.

**Comment by akoserwal** [comment]
Adding a flag would be helpful to set different configuration for the client.

**Pull request closed without merging**